### ISSUE TITLE
Maven repository distribution

### ISSUE THREAD

**Issue by GoogleCodeExporter** [anne/google-gin#45]
```
Please distribute this great library in a maven repository.

See also:
http://groups.google.com/group/google-gin/browse_thread/thread/211fa264a1b42b3a/
470f2b4c9b533ea2?lnk=gst&q=maven#470f2b4c9b533ea2

```

Original issue reported on code.google.com by `achim.hu...@googlemail.com` on 12 Jun 2009 at 4:23

**Comment by GoogleCodeExporter** [comment]
Original comment by `aragos` on 30 Jul 2009 at 6:36
- Added labels: **Priority-Low**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
I'm going to provide a server with a nightly build of Google-Gin.

I can add a maven repository to it, but I need instructions, as I don't know 
much about Maven.
```

Original comment by `terciofi...@gmail.com` on 19 Aug 2009 at 6:25

**Comment by GoogleCodeExporter** [comment]
```
The nightly build is up:

http://www.0x3333.com.br/builds/google-gin/

Maven soon.
```

Original comment by `terciofi...@gmail.com` on 21 Aug 2009 at 10:55

**Comment by GoogleCodeExporter** [comment]
```
using http://code.google.com/p/teraform-mvn/

    svn export http://google-gin.googlecode.com/svn/trunk/ google-gin-read-only
   mkdir google-gin
   mv google-gin-read-only google-gin/gin
   cd google-gin/
   CLEAN=1 DEBUG=1 ~/terraform_mvn.sh gin
   rm -fr gin/

review the attached sample for a fit.  I'm just learning gwt/gin and not really 
fully mavenized with a gwt 
project test bed yet, but this script should hit darn close with 
non-astonishing java projects.

```

Original comment by `northrup...@gmail.com` on 22 Aug 2009 at 9:52

Attachments:
- [google-gin-mvn.tgz](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-4/google-gin-mvn.tgz)

**Comment by GoogleCodeExporter** [comment]
```
This would be very useful to have, currently Gin is the only Maven dependency I 
have
which is not in the public maven repo.
```

Original comment by `bens%fas...@gtempaccount.com` on 10 Sep 2009 at 4:38

**Comment by GoogleCodeExporter** [comment]
```
Sorry guys, I'm without time to learn Maven right now, If someone could provide 
a maven repository in a tar.gz 
fashion I can update it and provide in my webserver.
```

Original comment by `terciofi...@gmail.com` on 10 Sep 2009 at 11:40

**Comment by GoogleCodeExporter** [comment]
```
Helo,
 IMHO the simplest solution could be to mavenize the project (still work with ant).

As a poc I attach:
 * a pom file to build the gin jar (class + java)
 * the "repository fragment" generated from the trunk.

HIH


 #mvn clean deploy.

```

Original comment by `olivier....@gmail.com` on 10 Sep 2009 at 2:11

Attachments:
- [pom.xml](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-7/pom.xml)
- [gin-mvn-1.0.0-SNAPSHOT.zip](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-7/gin-mvn-1.0.0-SNAPSHOT.zip)

**Comment by GoogleCodeExporter** [comment]
```
Hello,
any progress in this issue? would be greate to have GIN as Maven dependency. I 
see 
there is a PoC, so it should be no problem to move mavenize it.

TM
```

Original comment by `tomek.ma...@gmail.com` on 5 Nov 2009 at 2:49

**Comment by GoogleCodeExporter** [comment]
```
Hi!

My server is available, but I need 100% instructions how to create the 
repository.

Regards
```

Original comment by `terciofi...@gmail.com` on 6 Nov 2009 at 2:45

**Comment by GoogleCodeExporter** [comment]
```
Hi, those files below must be adapted with valid url/login/pwd...
1) add/adapt an distributionManagement section in the pom.xml (see first 
attachment).
2) create/modify/adapt $HOME/.m2/settings.xml (see second attachment)
the mvn clean deploy -Dmaven.test.skip=true should be ok
PS: I don't have the time right now to investigate the error that test case are 
running.
Best Regards.
```

Original comment by `olivier....@gmail.com` on 6 Nov 2009 at 3:39

Attachments:
- [pom.xml](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-10/pom.xml)
- [settings.xml](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-10/settings.xml)

**Comment by GoogleCodeExporter** [comment]
```
I would suggest a simpler path: don't move to a Maven build, just add a pom.xml 
which 
properly declares the dependencies ( gwt/guice ) . That's a small xml file 
which 
won't do anyone any harm :-)

Once that is done, a manual upload request can be submitted ( 
http://maven.apache.org/guides/mini/guide-central-repository-upload.html ).

If I get a nod of approval from one of the developers, I can create the POM 
myself 
and attach it here.


```

Original comment by `robert.munteanu` on 7 Jan 2010 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
Sounds great to me. Thanks!
```

Original comment by `bsto...@google.com` on 7 Jan 2010 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
Here's my initial pom file. It should be good enough for central, and I've 
tested it 
with a sample maven GWT project, with versions 1.5.3, 1.6.4, 1.7.0 and 2.0.0 .

Some notes:

- I declare a dependency on GWT 1.5.3, since 1.5 is the oldest supported 
version. 
That will simply be the default GWT version, and not fixed, since the user will 
most 
likely add GWT before Gin, and the GWT declared version will be picked
- I've added developer/connection SCM entries under tags/1.0.0 , since this is 
expected to be uploaded for version 1.0.0

Testing:

- Download the attached pom file;
- Download gin-1.0.zip and extract gin-1.0.jar
- Run mvn install:install-file -Dfile=gin-1.0.jar -DpomFile=pom.xml
- Include the following dependency in your POM file:

        <dependency>
            <groupId>com.google.gwt.inject</groupId>
            <artifactId>gin</artifactId>
            <version>1.0.0</version>
        </dependency>

You should be good to go now. Feedback is welcome, since I've only tested on a 
sample 
project.

As for getting into Maven central, it may require a few iterations on the POM - 
can't 
say for sure since this would be my first submission.
```

Original comment by `robert.munteanu` on 7 Jan 2010 at 10:11

Attachments:
- [pom.xml](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-13/pom.xml)

**Comment by GoogleCodeExporter** [comment]
```
Just a little comment for robert. The lib folder
(http://code.google.com/p/google-gin/source/browse/trunk/lib) in the svn repo
contains the dependencies needed for Gin. There is another library (aside from 
guice)
namely aopalliance.jar that is not mentioned in the pom You submitted. I'm no 
expert
(maybe it was used at one point and it isn't used anymore) but I think it 
should be
included in the pom.
```

Original comment by `dilbert....@gmail.com` on 8 Jan 2010 at 7:33

**Comment by GoogleCodeExporter** [comment]
```
@dilbert.elbonia: aopalliance.jar is a transitive dependency of guice 2.0 ( I 
double 
checked ) so we should be good here.
```

Original comment by `robert.munteanu` on 8 Jan 2010 at 7:35

**Comment by GoogleCodeExporter** [comment]
```
@robert: Thanks for the explanation. I forgot that the great thing about maven 
is
that you do not have to worry too much about transitive dependencies.
```

Original comment by `dilbert....@gmail.com` on 8 Jan 2010 at 7:49

**Comment by GoogleCodeExporter** [comment]
```
Also the scm connection is wrong, it should be in maven format. I'll update 
that as 
well.
```

Original comment by `robert.munteanu` on 8 Jan 2010 at 8:29

**Comment by GoogleCodeExporter** [comment]
```
Minor updates:

- Proper scm declaration;
- removed a redundant type=jar declaration.
```

Original comment by `robert.munteanu` on 8 Jan 2010 at 1:32

Attachments:
- [pom.xml](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-18/pom.xml)

**Comment by GoogleCodeExporter** [comment]
```
Bundle to be submitted to central.
```

Original comment by `robert.munteanu` on 15 Jan 2010 at 10:40

Attachments:
- [gin-1.0.0-bundle.jar](https://storage.googleapis.com/google-code-attachments/google-gin/issue-45/comment-19/gin-1.0.0-bundle.jar)

**Comment by GoogleCodeExporter** [comment]
```
Submitted at MAVENUPLOAD-2717: Please upload gin 1.0.0.
http://jira.codehaus.org/browse/MAVENUPLOAD-2717
```

Original comment by `robert.munteanu` on 15 Jan 2010 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
Gin is now on central.

http://repo1.maven.org/maven2/com/googlecode/gwt/inject/gin/1.0/



```

Original comment by `robert.munteanu` on 4 Feb 2010 at 1:53

**Comment by GoogleCodeExporter** [comment]
```
Thank You Robert!
```

Original comment by `goran.st...@gmail.com` on 4 Feb 2010 at 1:59

**Comment by GoogleCodeExporter** [comment]
```
Well done!
```

Original comment by `frode.re...@gmail.com` on 4 Feb 2010 at 2:09

**Comment by GoogleCodeExporter** [comment]
```
Nice one
```

Original comment by `tband...@gmail.com` on 10 Mar 2010 at 10:13

**Comment by GoogleCodeExporter** [comment]
```
Thanks for getting this done! 

Before I mark this issue as successfully completed, I'd like to add a link/hint
somewhere on the page and knowing nothing about Maven: Can you tell me how that
should look? Do I just point developers to
http://repo1.maven.org/maven2/com/googlecode/gwt/inject/gin/1.0/?
```

Original comment by `aragos` on 11 Mar 2010 at 11:52
- Changed state: **Started**
- Added labels: **Type-Task**
- Removed labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
```
<dependency>
  <groupId>com.googlecode.gwtinject</groupId>
  <artifactId>gin</artifactId>
  <version>1.0</version>
</dependency>

This is, how you use it.
```

Original comment by `jan.ehrhardt` on 12 Mar 2010 at 5:09

**Comment by GoogleCodeExporter** [comment]
```
The dependency posted above is (almost) correct.  The groupId is 
com.googlecode.gwt.inject , so the xml fragment should be 

        <dependency>
            <groupId>com.googlecode.gwt.inject</groupId>
            <artifactId>gin</artifactId>
            <version>1.0</version>
        </dependency>


There is no other setup necessary since the jars and all its dependencies are 
available on Maven central.
```

Original comment by `robert.munteanu` on 12 Mar 2010 at 7:21

**Comment by GoogleCodeExporter** [comment]
```
Thanks all for your help!
```

Original comment by `aragos` on 12 Mar 2010 at 10:47
- Changed state: **Done**

**Comment by GoogleCodeExporter** [comment]
```
What about 1.5 version?
```

Original comment by `helper.h...@gmail.com` on 22 Feb 2011 at 12:58

**Comment by GoogleCodeExporter** [comment]
```
I'll set some time aside this week. Stay tuned for a call for testing on the 
gin mailing list.
```

Original comment by `robert.munteanu` on 22 Feb 2011 at 1:00

**Comment by GoogleCodeExporter** [comment]
```
Note for readers: re mvn dependency, the 1.0 version is available in two 
locations:
http://repo2.maven.org/maven2/com/google/gwt/inject/gin/1.0/
http://repo2.maven.org/maven2/com/googlecode/gwt/inject/gin/1.0/
... thus, two different groupId values, which is quite confusing.

Thanks for pushing the 1.5.0 version:
http://repo2.maven.org/maven2/com/google/gwt/inject/gin/1.5.0/

Thus the <groupId>com.google.gwt.inject</groupId> is valid.
```

Original comment by `JasonThr...@gmail.com` on 5 Mar 2011 at 7:09

**Comment by GoogleCodeExporter** [comment]
```
The GIN homepage links to comment #27.  Can we get some offical @google.com 
ruling on which groupID to use?
```

Original comment by `sne...@gmail.com` on 14 Mar 2011 at 6:58

**Comment by GoogleCodeExporter** [comment]
```
The version uploaded by Google is at 
http://repo2.maven.org/maven2/com/google/gwt/inject/gin/1.5.0/
```

Original comment by `drfibonacci@google.com` on 14 Mar 2011 at 7:10

**Comment by GoogleCodeExporter** [comment]
```
I'm Using Roo 1.1.5

Open pom.xml file from your roo projects and than replace : 

        <dependency>
            <groupId>com.googlecode.gwt.inject</groupId>
            <artifactId>gin</artifactId>
            <version>1.5</version>
            <classifier/>
        </dependency>

to 

        <dependency>
            <groupId>com.google.gwt.inject</groupId>
            <artifactId>gin</artifactId>
            <version>1.5.0</version>
            <classifier/>
        </dependency>

it's works fine for me
```

Original comment by `alrasyid...@gmail.com` on 5 Dec 2011 at 11:49

**Comment by GoogleCodeExporter** [comment]
```
The newest version is 2.0.0. You'll want to depend on Gin like this:

 <dependency>
            <groupId>com.google.gwt.inject</groupId>
            <artifactId>gin</artifactId>
            <version>2.0.0</version>
            <classifier/>
</dependency>
```

Original comment by `aragos` on 27 Sep 2012 at 7:10

**Comment by GoogleCodeExporter** [comment]
```
Can we bump up the version? Now when injection order is fixed it would be nice 
to have at least sth like 2.0.1. That would save a lot of hassle 
```

Original comment by `przemek....@gmail.com` on 4 Jun 2013 at 11:32
